chromium/background.js

```diff
@@ -361,12 +363,18 @@ function onBeforeRequest(details) {
     }
   }
 
-  if (newuristr && using_credentials_in_url) {
-    // re-insert userpass info which was stripped temporarily
-    const uri_with_credentials = new URL(newuristr);
-    uri_with_credentials.username = tmp_user;
-    uri_with_credentials.password = tmp_pass;
-    newuristr = uri_with_credentials.href;
+  if (using_credentials_in_url) {
+    const canonical_url_with_credentials = new URL(canonical_url);
+    canonical_url_with_credentials.username = tmp_user;
+    canonical_url_with_credentials.password = tmp_pass;
+    canonical_url = canonical_url_with_credentials.href;
```

Comment:
It should be safe to add this as an `else` clause to the conditional immediately below.  If `newuristr` is defined, this mutation of `canonical_url` is simply unused.

This is a minor point and shouldn't block merging, but it makes things a bit easier to follow with the matching `if (!newuristr)` below.